code/processes/dqe.q

```diff
@@ -1,34 +1,27 @@
 \d .dqe
 
-/- default parameters
-subscribeto:@[value;`subscribeto;`];                          /-list of tables subscribed to
-subscribesyms:@[value;`subscribesyms;`];                      /-list of syms subscribed to
-schema:@[value;`schema;0b];                                   /-retrieve schema from tickerplant
-tickerplanttypes:@[value;`tickerplanttypes;`tickerplant];     /-list of tickerplant types to try and make a connection to
-replaylog:@[value;`replaylog;0b];                             /-replay the tickerplant log file
-tpconnsleepintv:@[value;`tpconnsleepintv;10];                 /-default wait time before trying to reconnect to discovery/TP
-/- end of default parameters
-
-subscribe:{[]
-  if[count s:.sub.getsubscriptionhandles[tickerplanttypes;();()!()];
-  .lg.o[`subscribe;"found available tickerplant, attempting to subscribe"];
-  /-set the date that was returned by the subscription code i.e. the date for the tickerplant log file
-  /-and a list of the tables that the process is now subscribing for
-  subinfo: .sub.subscribe[subscribeto;subscribesyms;schema;replaylog;first s];
-  /-setting subtables and tplogdate globals
-  .dqe,:subinfo]
-  }
-
 init:{
-  .servers.CONNECTIONS:distinct .servers.CONNECTIONS, .dqe.tickerplanttypes;
   .lg.o[`init;"searching for servers"];
-  .servers.startup[];
-  .servers.startupdependent[tickerplanttypes;tpconnsleepintv];
-  subscribe[];
+  .servers.startup[];                                                                                           /- Open connection to discovery
   }
 
-tableExists:{x in tables[]};                                  /-function to check for table, param is table name as a symbol
+tableexists:{x in tables[]};                                                                                    /- function to check for table, param is table name as a symbol
+
+runcheck:{[fn;vars;rs]                                                                                          /- function used to send other function to test processes
+  discov:.servers.querydiscovery`ALL;
+  if[0<count s:{
+      $[1<count sf:x in raze exec procname,proctype from y;x where 0=sf;(enlist x) where 0=enlist sf]           /- check if processes exist and are valid, slight diff check if atom or list input
```

Comment:
Is this line just about working with lists? And if so will this go in its place?
```suggestion
      x where not sf:x in raze exec procname,proctype from y          /- check if processes exist and are valid, slight diff check if atom or list input
```

Reply:
I has to work with atoms too as a single process can be passed in to be checked.
Example:
```
.dqe.runcheck[.dqe.tableexists;`quote;`rdb]
```
There is a problem with the where returning a type error when a list is not passed to it.

Reply:
Can you just use `(),` to make it a list?

```q
q)(),`rdb
,`rdb
q)(),`rdb`hdb
`rdb`hdb
```